addons/stock/models/stock_move.py

```diff
@@ -1293,7 +1293,7 @@ def _action_assign(self):
             rounding = roundings[move]
             missing_reserved_uom_quantity = move.product_uom_qty - reserved_availability[move]
             missing_reserved_quantity = move.product_uom._compute_quantity(missing_reserved_uom_quantity, move.product_id.uom_id, rounding_method='HALF-UP')
-            if move._should_bypass_reservation():
+            if move._should_bypass_reservation() and not (move.product_id.type == 'consu' and move.move_orig_ids.filtered(lambda m: m.state == 'done').mapped('move_line_ids.result_package_id')):
```

Comment:
Should be done in the `_should_bypass_reservation` method ? It created inconsistency between the method `_should_bypass_reservation` and the reservation it self.
Also I am concern about the performance cost of this.

Reply:
Performance:
Yeah, I have no idea, maybe e empiric test is needed with the populate tool (I will try it). I am not sure that call it twice is problematic since the cache should do the job.

In `_should_bypass_reservation`:
the method is also highly call in the `_action_done` of SML and the `write`. I am not sure that it is correct in these case